test/fakeintake/aggregator/processAggregator_test.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	_ "embed"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed fixtures/process_payload_bytes
+var processPayload []byte
```

Comment:
any way to better document the structure of the payload? Mostly thinking about what would need to happen if we wanted to change it. One idea might be to have the payload file in plain JSON, and then the test would serialize it to the relevant binary format

Reply:
I was more thinking about the case where we want to add a new field to the test payload while using the existing proto definition (i.e. just changing the test input), or changing an existing field in the test payload

Reply:
I've added the test payload as a go struct and encoded it during the tests. That way, any changes can be made to the human-readable struct.